Sources/TSCUtility/PolymorphicCodable.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// Allows encoding and decoding known polymorphic types.
+public protocol PolymorphicCodableProtocol: Codable {
+    static var implementations: [PolymorphicCodableProtocol.Type] { get }
```

Comment:
Could be a Set<> I suppose 🤔

Reply:
Can't use Set:

Type 'PolymorphicCodableProtocol.Type' does not conform to protocol 'Hashable' and Cannot extend a metatype 'PolymorphicCodableProtocol.Type'

Reply:
Since `String(reflecting: )` is needed every time `implementations` is accessed, it might make sense to do a struct representing the `(String, Type)`, and implement a `Hashable` over it. Then it is going to be `Set`-compatible. I am sure you've thought of that, just 2c.